dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/inmemory/compiler/InMemoryJDTCompiler.java

```diff
@@ -123,8 +126,12 @@ public JspCompilerResult compile(JspResources[] jspResources, JspResources[] dep
         }
         compilerOptionsMap.put(CompilerOptions.OPTION_Encoding, javaEncoding);
         
-        //487396.1 jdkSourceLevel is 15 by default now ... should get into if statement
-        if (jdkSourceLevel == 14) {
+        if (javaSourceLevel != -1) {
+            compilerOptionsMap.put(CompilerOptions.OPTION_Source, javaSourceLevel.toString());
+            compilerOptionsMap.put(CompilerOptions.OPTION_Compliance, javaSourceLevel.toString());
+            compilerOptionsMap.put(CompilerOptions.OPTION_TargetPlatform, javaSourceLevel.toString());
+        }
+        else if (jdkSourceLevel == 14) {
```

Comment:
InMemory Compilation is [broken](https://github.com/OpenLiberty/open-liberty/issues/15240), so I cannot test this change.